### PR TITLE
fix for mendix 9

### PR DESCRIPTION
**Pull request by JansenNick**
Previously we used the attributes values for the Options also for the DefaultValue. In mendix 9 this is not possible anymore, so they had to be added separately.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "customdropdown",
   "widgetName": "CustomDropdown",
-  "version": "1.0.2",
+  "version": "2.0.0",
   "description": "Widget to make a selection in a dropdown menu.",
   "copyright": "2021 Mendix Technology BV",
   "author": "Nick",
```
Comment (reinouts) [inline package.json:7]: Feel free to put your full name there ;)

File: src/components/CustomDropdownComp.tsx [hunk 2]
```diff
@@ -43,54 +47,82 @@ export default function CustomDropdownComp(props: CustomDropdownComponentProps):
...
         value: label,
+        // eslint-disable-next-line object-shorthand
         id: id,
         secondLabel: secondaryLabel,
         url: urlstring
     });
 
     useEffect(() => {
-        if (props.defaultValue.status === 'available') {
+        if (props.defaultValue.status === "available") {
```
Comment (dtitenko) [inline src/components/CustomDropdownComp.tsx:67]:
```suggestion
        if (props.defaultValue.status === ValueStatus.Available) {
```
That is also in [my PR](https://github.com/mendixlabs/CustomDropdown/pull/6/files#diff-15bf84ebcfbd6c361601d0f236a7e9a25bf33bbdd14e5cff8fc6048615c89dbdR40) which is still open ((((((

File: src/components/CustomDropdownComp.tsx [hunk 3]
```diff
@@ -43,54 +47,82 @@ export default function CustomDropdownComp(props: CustomDropdownComponentProps):
...
+            <Label
+                DisplayName={label}
+                UrlString={urlstring}
+                ClassNamePrefix={props.classNamePrefix}
+                EnableAvatar={props.useAvatar}
+                SecondLabel={secondaryLabel}
+            />
+        ),
         value: label,
+        // eslint-disable-next-line object-shorthand
```
Comment (dtitenko) [inline src/components/CustomDropdownComp.tsx:60]: Why? Just leave id. What it wants is this
```
{
   value: label,
   id,
   secondLabel,
   url: urlstring
}
```

File: src/components/CustomDropdownComp.tsx [hunk 4]
```diff
@@ -43,54 +47,82 @@ export default function CustomDropdownComp(props: CustomDropdownComponentProps):
...
+                    firstLabelDefaultValue,
+                    secondLabelDefaultValue,
+                    objIdDefaultValue,
+                    imgUrlDefaultValue
+                }: {
+                    firstLabelDefaultValue: string;
+                    secondLabelDefaultValue: string;
+                    objIdDefaultValue: string;
+                    imgUrlDefaultValue: string;
+                } = getLabelValuesDefault(obj);
```
Comment (dtitenko) [inline src/components/CustomDropdownComp.tsx:79]: make non-exported (private) interface above prop interface. Now it's barely readable.

File: src/components/CustomDropdownComp.tsx [hunk 5]
```diff
@@ -43,54 +47,82 @@ export default function CustomDropdownComp(props: CustomDropdownComponentProps):
...
+                    firstLabelOptions,
+                    secondLabelOptions,
+                    objIdOptions,
+                    imgUrlOptions
+                }: {
+                    firstLabelOptions: string;
+                    secondLabelOptions: string;
+                    objIdOptions: string;
+                    imgUrlOptions: string;
+                } = getLabelValuesOption(obj);
```
Comment (dtitenko) [inline src/components/CustomDropdownComp.tsx:108]: Here you'll use same interface.

File: src/components/CustomDropdownComp.tsx [hunk 6]
```diff
@@ -43,54 +47,82 @@ export default function CustomDropdownComp(props: CustomDropdownComponentProps):
...
+                return createOption(firstLabelOptions, secondLabelOptions, objIdOptions, imgUrlOptions);
+            });
             setOptions(options);
         }
     }, [props.options]);
 
     const handleChange = (inputValue: any, actionMeta: any) => {
         switch (actionMeta.action) {
-            case 'select-option': {
+            case "select-option": {
```
Comment (dtitenko) [inline src/components/CustomDropdownComp.tsx:117]: Check for enum in `select` typings, I'm sure it is there. If not, make one.

**Review by dtitenko — CHANGES_REQUESTED** [review]

File: src/components/CustomDropdownComp.tsx (hunk 3 above)
Comment (JansenNick) [reply src/components/CustomDropdownComp.tsx:60]: Ah yes I'm not very familiar with that notation shorthand, so it confused me a bit. Will update it

**Review by dtitenko — APPROVED** [review]

**Pull request merged**